sklearn/ensemble/_gb.py

```diff
@@ -369,10 +375,14 @@ def _resize_state(self):
                 self.oob_improvement_ = np.resize(
                     self.oob_improvement_, total_n_estimators
                 )
+                self.oob_scores_ = np.resize(self.oob_scores_, total_n_estimators)
+                self.oob_score_ = self.oob_scores_[-1]
```

Comment:
Here, it will be potentially a random number while it will be initialized at `0.0` in the case below.
Let's initialize to `np.nan`. At least we can detect that something went wrong (which I am not sure that we can trigger).